### PR TITLE
docs(api): enhance `abortNavigation` util

### PR DESCRIPTION
**Pull request by Krutie**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add examples to API documentation for `abortNavigation` utility function

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.api/3.utils/abort-navigation.md [hunk 1]
```diff
@@ -1,16 +1,66 @@
...
+You can pass the error as an `Error` object that includes the error code and a message.
+
+```ts [middleware/auth.ts]
+export default defineNuxtRouteMiddleware((to, from) => {
+ const auth = useState('auth')
+ if (to.params.id === '1') {
+    abortNavigation({ 
+       statusCode: 401, 
+       statusMessage: "This feature requires special permission." 
+  })
```
Comment (danielroe) [inline docs/content/3.api/3.utils/abort-navigation.md:60]:
```suggestion
       statusMessage: 'This feature requires special permission.'
    })
```

File: docs/content/3.api/3.utils/abort-navigation.md [hunk 2]
```diff
@@ -1,16 +1,66 @@
...
 `abortNavigation()` is only usable inside a [route middleware handler](/guide/directory-structure/middleware).
 ::
 
 Inside a route middleware handler, `abortNavigation()` will abort navigation, and throw an error if one is set as a parameter.
 
+## Examples
+
+The example below shows how you can use `abortNavigation` in route middleware to prevent unauthorised route access.
+
+```jsx
```
Comment (pi0) [inline docs/content/3.api/3.utils/abort-navigation.md:25]:
```suggestion
```ts
```

File: docs/content/3.api/3.utils/abort-navigation.md [hunk 3]
```diff
@@ -1,16 +1,66 @@
...
+// middleware/auth.ts
+export default defineNuxtRouteMiddleware((to, from) => {
+ const user = useState('user')
+ if (!user.value.isAuthorized) {
+    abortNavigation()
```
Comment (pi0) [inline docs/content/3.api/3.utils/abort-navigation.md:30]: We should either throw or await or return.

File: docs/content/3.api/3.utils/abort-navigation.md [hunk 4]
```diff
@@ -1,16 +1,68 @@
...
+
+The example below shows how you can use `abortNavigation` in a route middleware to prevent unauthorized route access:
 
-Inside a route middleware handler, `abortNavigation()` will abort navigation, and throw an error if one is set as a parameter.
+```ts [middleware/auth.ts]
+export default defineNuxtRouteMiddleware((to, from) => {
+  const user = useState('user')
+
+  if (!user.value.isAuthorized) {
+    abortNavigation()
```
Comment (DamianGlowala) [inline docs/content/3.api/3.utils/abort-navigation.md:32]:
```suggestion
    return abortNavigation()
```

File: docs/content/3.api/3.utils/abort-navigation.md (hunk 3 above)
Comment (DamianGlowala) [reply docs/content/3.api/3.utils/abort-navigation.md:32]: `abortNavigation` either throws an error or returns `false`, can it be awaited? Shall we go for returning (for consistency with `return navigateTo('/edit-post')`)?

**Review by DamianGlowala — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks.

**Pull request merged**